treeherder/perf/models.py

```diff
@@ -256,6 +299,8 @@ class Meta:
         unique_together = ('repository', 'job', 'push', 'signature')
 
     def save(self, *args, **kwargs):
+        if not self.id:
+            self.id = self.next_id()
```

Comment:
Do I understand that this is to handle older rows?

Reply:
There are many instances of `PerformanceDatum.objects.create()`, none of which provide an `id`; they depend on the database `AUTO INCREMENT` to provide that.  This code replaces that feature: If there is no `id` then set it to a unique value.  For objects that already exist, the `id` already exists, and should not change.